cdap-hbase-compat-1.1/src/main/java/co/cask/cdap/data2/increment/hbase11/IncrementSummingScanner.java

```diff
@@ -0,0 +1,379 @@
+    }
+    throw new IllegalStateException(
+      "RegionScanner.isFilterDone() called when the wrapped scanner is not a RegionScanner");
+  }
+
+  @Override
+  public boolean reseek(byte[] bytes) throws IOException {
+    if (baseRegionScanner != null) {
+      return baseRegionScanner.reseek(bytes);
+    }
+    throw new IllegalStateException(
+      "RegionScanner.reseek() called when the wrapped scanner is not a RegionScanner");
+  }
+
+  @Override
+  public long getMaxResultSize() {
+    if (baseRegionScanner != null) {
+      return baseRegionScanner.getMaxResultSize();
+    }
+    throw new IllegalStateException(
+      "RegionScanner.isFilterDone() called when the wrapped scanner is not a RegionScanner");
+  }
+
+
+  @Override
+  public long getMvccReadPoint() {
+    if (baseRegionScanner != null) {
+      return baseRegionScanner.getMvccReadPoint();
+    }
+    throw new IllegalStateException(
+      "RegionScanner.isFilterDone() called when the wrapped scanner is not a RegionScanner");
+  }
+
+  /**
+   * @return The limit on the number of cells to retrieve on each call to next(). See
+   * {@link Scan#setBatch(int)}
+   */
+  @Override
+  public int getBatch() {
+    return batchSize;
+  }
+
+  @Override
+  public boolean nextRaw(List<Cell> cells) throws IOException {
+    return nextRaw(cells, ScannerContext.newBuilder().setBatchLimit(batchSize).build());
+  }
+
+  @Override
+  public boolean nextRaw(List<Cell> cells, ScannerContext scannerContext) throws IOException {
+    return nextInternal(cells, scannerContext);
+  }
+
+  @Override
+  public boolean next(List<Cell> cells) throws IOException {
+    return next(cells, ScannerContext.newBuilder().setBatchLimit(batchSize).build());
+  }
+
+  @Override
+  public boolean next(List<Cell> cells, ScannerContext scannerContext) throws IOException {
+    return nextInternal(cells, scannerContext);
+  }
+
+  private boolean nextInternal(List<Cell> cells, ScannerContext scannerContext) throws IOException {
+    if (LOG.isTraceEnabled()) {
+      LOG.trace("nextInternal called with limit=" + scannerContext);
+    }
+    Cell previousIncrement = null;
+    long runningSum = 0;
+    int addedCnt = 0;
+    baseScanner.startNext();
+    Cell cell;
+    // We need to to access the batch limit (limit on number of columns) in the ScannerContext.
+    // The ScannerContext does not exposes its method to get the limits so we have a Reader under the same package
+    // which can supply us these limits given a ScannerContext. Also, we cannot depend on the limit tracking of the
+    // internal scanner as it does not differentiate between an increment cell and a non-increment cell.
+    // So, we read all the cells from the internal scanner in batches of limit and do the tracking for columns here
+    // and stop when we have reached the limit.
+    ScannerContextReader scannerContextReader = new ScannerContextReader(scannerContext);
+    int limit = scannerContextReader.getBatchLimit();
```

Comment:
It would be a good idea to move the above lines into a private method called `getBatchLimit()` or something. Then the six lines of comments can also move to the private method.